chart/templates/secrets/git-ssh-key-secret.yaml

```diff
@@ -0,0 +1,34 @@
+{{/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+*/}}
+
+{{- if and .Values.dags.gitSync.gitSshKey .Values.dags.gitSync.enabled}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}
+    heritage: {{ .Release.Service }}
+    {{- with .Values.labels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+  name: airflow-ssh-secret
+data:
+  gitSshKey: {{ .Values.dags.gitSync.gitSshKey | quote }}
```

Comment:
At least so far, we've accepted non-base64'd secret values, like webserver secret key. Is there a reason to require base64'd values for this?

e.g. https://github.com/apache/airflow/blob/93ba263732a30ed51fdae0b2bff7cd959be115b8/chart/templates/secrets/webserver-secret-key-secret.yaml#L40

Reply:
I assume it from the [doc](https://github.com/apache/airflow/blob/main/docs/helm-chart/manage-dags-files.rst#mounting-dags-from-a-private-github-repo-using-git-sync-sidecar) but reverted it like you said

Reply:
This is still expecting a b64 string?

Reply:
We supposed to get here 64b string but we don't convert it now

Reply:
Let me rephrase. Today, afaik, all our secret values are not base64 encoded in the values file. Is there a reason the sshKey needs to be encoded in the values file (say, newlines or something), or can this work (completely untested):

```
dags:
  gitSync:
    sshKey: |-
      -----BEGIN OPENSSH PRIVATE KEY-----
      ...
      -----END OPENSSH PRIVATE KEY-----
```

Reply:
This will work.
So the value not have to be base64 :)

Reply:
No we dont 😟

Reply:
Here is what I see. In the secret:

```
data:
  gitSshKey: {{ .Values.dags.gitSync.sshKey | quote }}
```

And in the values file we have:

```
    # Or set sshKeySecret with your key
    # sshKey: <base64-converted-ssh-private-key>
```

That's why I say we are still expecting the user to put in a base64'd value - Secrets `data` takes base64 values, and it's documented as such in the values file.

Reply:
I'm just saying it's inconsistent with the other secret values we gather from the user. If there isn't a technical reason to have them b64 it before hand, we should probably be consistent and take the raw value.

Reply:
I changed it, they don't have to be b64 before hand.

Reply:
https://github.com/apache/airflow/pull/39936#discussion_r1643492238

> Here is what I see. In the secret:
> 
> ```
> data:
>   gitSshKey: {{ .Values.dags.gitSync.sshKey | quote }}
> ```
> 
> And in the values file we have:
> 
> ```
>     # Or set sshKeySecret with your key
>     # sshKey: <base64-converted-ssh-private-key>
> ```
> 
> That's why I say we are still expecting the user to put in a base64'd value - Secrets `data` takes base64 values, and it's documented as such in the values file.

I totally missed this comment.
I got these comment from the doc here:
https://github.com/apache/airflow/blob/1f7be6f63e48d1861f913f2f5449a7a9e1bfd1fb/docs/helm-chart/manage-dags-files.rst?plain=1#L263
But I think it can get a raw value, therefore I removed the comment says it's need to be base64.